File: Door2Door-Web/src/app/model/Room.ts
import {RoomType} from "./RoomType";
import { GeoJsonObject } from 'geojson';
import { PointOfEntry } from "./PointOfEntry";
import { Geometry } from "./Geometry";

export class Room extends Geometry {

  constructor(
    // Scalar fields.
    public id: Number,
    public name: string,
    public type: RoomType,
    public geometry: GeoJsonObject,
    ) {
    super(geometry);
  }

}

File: Door2Door-API/Room.cs
using GeoJSON.Net;
using GeoJSON.Net.Geometry;

namespace Door2Door_API;

public class Room
{
    public long Id { get; set; }
    public string? Geometry { get; set; }
    public RoomType RoomType { get; set; }
    public string RoomName { get; set; }
}